controller/array_action/array_mediator_svc.py

```diff
@@ -443,29 +479,45 @@ def _stop_and_delete_fcmap(self, fcmap_id):
         self._stop_fcmap(fcmap_id)
         self._delete_fcmap(fcmap_id, force=True)
 
-    def _delete_snapshot(self, target_cli_volume):
-        self._stop_and_delete_fcmap(target_cli_volume.FC_id)
-        self._delete_volume_by_name(target_cli_volume.name)
+    def _delete_all_fcmaps_as_source_if_exist(self, snapshot_name):
+        snapshot_as_source_fcmaps = self._get_fcmaps(snapshot_name, ENDPOINT_TYPE_SOURCE)
+        unfinished_fcmaps = [fcmap.name for fcmap in snapshot_as_source_fcmaps
+                             if fcmap.status != FCMAP_STATUS_DONE]
+        if unfinished_fcmaps:
+            raise controller_errors.SnapshotIsStillInUseError(snapshot_name, unfinished_fcmaps)
+        for fcmap in snapshot_as_source_fcmaps:
+            self._delete_fcmap(fcmap.id, force=False)
```

Comment:
no need to stop? ( self._stop_fcmap)

Reply:
no, since they should all be finished (same reason for `force=False`)

Reply:
ok, I dont understand svc so I have no really idea what it means..